tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/PagedMirrorTest.java

```diff
@@ -0,0 +1,161 @@
+      Assert.assertNotNull(snf2);
+
+      File countJournalLocation = server1.getConfiguration().getJournalLocation();
+      Assert.assertTrue(countJournalLocation.exists() && countJournalLocation.isDirectory());
+      String protocol = "amqp";
+
+      ConnectionFactory sendCF = CFUtil.createConnectionFactory(protocol, sendURI);
+      ConnectionFactory consumeCF = CFUtil.createConnectionFactory(protocol, consumeURI);
+      ConnectionFactory secondConsumeCF = CFUtil.createConnectionFactory(protocol, secondConsumeURI);
+
+      String bodyBuffer;
+      {
+         StringBuffer buffer = new StringBuffer();
+         for (int i = 0; i < 1024; i++) {
+            buffer.append("*");
+         }
+         bodyBuffer = buffer.toString();
+      }
+
+      int NUMBER_OF_MESSAGES = 200;
+      int ACK_I = 77;
+
+      try (Connection sendConnecton = sendCF.createConnection()) {
+         Session sendSession = sendConnecton.createSession(true, Session.SESSION_TRANSACTED);
+         Queue jmsQueue = sendSession.createQueue("someQueue");
+         MessageProducer producer = sendSession.createProducer(jmsQueue);
+
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            TextMessage message = sendSession.createTextMessage(bodyBuffer);
+            message.setIntProperty("i", i);
+            producer.send(message);
+         }
+         sendSession.commit();
+      }
+
+      Wait.assertEquals(0, snf1::getMessageCount);
+      Wait.assertEquals(0, snf2::getMessageCount);
+
+      try (Connection consumeConnection = consumeCF.createConnection()) {
+         Session consumeSession = consumeConnection.createSession(false, 101); // individual ack
+         Queue jmsQueue = consumeSession.createQueue("someQueue");
+         MessageConsumer consumer = consumeSession.createConsumer(jmsQueue);
+         consumeConnection.start();
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            TextMessage message = (TextMessage) consumer.receive(6000);
+            if (message.getIntProperty("i") == ACK_I) {
+               message.acknowledge();
+            }
+         }
+         Assert.assertNull(consumer.receiveNoWait());
+      }
+      Wait.assertEquals(0, snf1::getMessageCount);
+      Wait.assertEquals(0, snf2::getMessageCount);
+      Wait.assertEquals(1, () -> acksCount(countJournalLocation), 5000, 1000);
+
+      try (Connection consumeConnection = secondConsumeCF.createConnection()) {
+         Session consumeSession = consumeConnection.createSession(true, Session.SESSION_TRANSACTED);
+         Queue jmsQueue = consumeSession.createQueue("someQueue");
+         MessageConsumer consumer = consumeSession.createConsumer(jmsQueue);
+         consumeConnection.start();
+
+         for (int i = 0; i < NUMBER_OF_MESSAGES - 1; i++) {
+            TextMessage message = (TextMessage) consumer.receive(6000);
+            Assert.assertNotNull(message);
+            Assert.assertNotEquals(ACK_I, message.getIntProperty("i"));
```

Comment:
Would be nicer to assert the others are as expected as opposed to just not-this-one. You could make a list and use one of the assertion/matcher libs that would compare an expected list at the end, e.g hamcrest is already in the build.

Reply:
My intention was to make sure this message was ack. We have other tests to validate the other cases.

Reply:
I disagree as the other tests dont appear to ack an individual message in the middle of the queue, and if thats the case then no other test actually can be verifying what I suggested to (essentially that behaviour for the rest of the queued messages is as expected ad not in some way broken).

All this checks is that the 199 messages then received arent the one it acked. It would still pass if the broker had sent completely the wrong messages, even the same message over and over, etc...its just not as robust a test as it could trivialy be.